lib/SILGen/SILGenExpr.cpp

```diff
@@ -3782,7 +3783,69 @@ visitMagicIdentifierLiteralExpr(MagicIdentifierLiteralExpr *E, SGFContext C) {
   llvm_unreachable("Unhandled MagicIdentifierLiteralExpr in switch.");
 }
 
-RValue RValueEmitter::visitCollectionExpr(CollectionExpr *E, SGFContext C) {
+RValue RValueEmitter::visitArrayExpr(ArrayExpr *E, SGFContext C) {
+  FormalEvaluationScope Scope(SGF);
+
+  auto loc = SILLocation(E);
+  auto init = E->getInitializer();
+  auto *decl = dyn_cast<AbstractFunctionDecl>(init.getDecl());
+
+  auto elementType = E->getElementType();
+
+  auto argLabels = decl->getFullName().getArgumentNames();
+
+  llvm::SmallVector<CleanupHandle, 5> cleanups;
+
+  auto arrayTy = ArraySliceType::get(elementType)->getCanonicalType();
+
+  // TODO: Support vararg expansion (VarargExpansionExpr) in array literals?
+  auto varargs_info =
+      emitBeginVarargs(SGF, loc, elementType->getCanonicalType(), arrayTy,
+                       E->getNumElements(), {});
+
+  for (unsigned Index = 0; Index < E->getNumElements(); ++Index) {
```

Comment:
Another example of inconsistent capitalization please fix. Also a nicer way of doing this is to use range:

```
for (unsigned index : range(E->getNumElements())) {
```